drivers/docker/driver.go

```diff
@@ -990,6 +990,13 @@ func (d *Driver) createContainerConfig(task *drivers.TaskConfig, driverConfig *T
 	if _, ok := d.config.allowRuntimes[containerRuntime]; !ok && containerRuntime != "" {
 		return c, fmt.Errorf("requested runtime %q is not allowed", containerRuntime)
 	}
+	if d.config.NewNetworking {
+		if containerRuntime == "" {
+			containerRuntime = "nomad"
+		} else {
+			return c, fmt.Errorf("new-style networking not compatible with custom runtimess")
```

Comment:
So that is just me being lazy. In practice, the docker api supports querying the registered runtimes (run `docker info` on your box and you will see them, so it must be part of the API). We could query that and then pass the "sub"-runtime through via annotations like with the network ns (we have to use annotations since it is docker calling the runtime and we cannot pass arbritrary args)

Reply:
Okay we cannot pass on the binary to call next via annotations since the oci runtime operations like `state` & `delete` only pass in a container ID. So the only thing we can do is force the enduser to specify something like this in docker daemon.json:

```json
{
	"runtimes": {
          "nomad": {"path": "/home/florian/sources/nomad/bin/nomad",
                             "runtimeArgs": ["runc", "/usr/bin/runc"]},
          "nomad-nvidia": {"path": "/home/florian/sources/nomad/bin/nomad",
                             "runtimeArgs": ["runc", "/usr/bin/nvidia-runtime"]},
       }
}
```

and then we can just prefix `nomad-` to the requested runtime if the "new" networking is enabled. Not really nice but probably good enough.

Reply:
Added a commit that prefixes other runtimes with `nomad-*`. Should we consider the interaction with `allow_runtimes`? I am leaning towards a no since those `nomad-*` runtimes have to be configured explicitly anyways?